subsys/net/ip/ipv4_fragment.c

```diff
@@ -519,6 +519,9 @@ int net_ipv4_send_fragmented_pkt(struct net_if *iface, struct net_pkt *pkt,
 
 	NET_PKT_DATA_ACCESS_DEFINE(frag_access, struct net_ipv4_hdr);
 	frag_hdr = (struct net_ipv4_hdr *)net_pkt_get_data(pkt, &frag_access);
+	if (!frag_hdr) {
+		return -ENOBUFS;
```

Comment:
It's not a buffer error but a contiguous error though? Though with that said, I don't think we have an error code to indicate that. This will cause it to retry sending and if it's not contiguous memory I don't see it ever being successful. Thoughts @rlubos?

Reply:
shouldn't it be similar to `send_ipv6_fragment()` ?

Reply:
Well, I think it should make sense in the first place - I don't like that either that data **access** error in IPv6 code is reported as `-ENOBUFS` (but there are also other differences in the IPv6/IPv4 logic - in IPv6 `ENOBUFS` results in a packet drop).